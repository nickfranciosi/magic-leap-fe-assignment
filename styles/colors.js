const alert = '#e15021';
const backgroundMain = '#151515';
const border = '#333';
const callout = '#9e4f60';
const callToAction = '#edec51';
const subduedText = '#ddd';
const primaryText = '#fff';

export default {
  alert,
  backgroundMain,
  border,
  callout,
  callToAction,
  subduedText,
  primaryText
};
