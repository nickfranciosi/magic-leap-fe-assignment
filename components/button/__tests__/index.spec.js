/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import PrimaryButton from '../primaryButton';
import RemoveButton from '../removeButton';
import colors from '../../../styles/colors';

describe('PrimaryButton', () => {
  test('it should render', () => {
    const tree = renderer.create(<PrimaryButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should be transparent', () => {
    const tree = renderer.create(<PrimaryButton />).toJSON();
    expect(tree).toHaveStyleRule('background-color', 'transparent');
  });
});

describe('RemoveButton', () => {
  test('it should render', () => {
    const tree = renderer.create(<RemoveButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should have alert color', () => {
    const tree = renderer.create(<RemoveButton />).toJSON();
    expect(tree).toHaveStyleRule('color', colors.alert);
  });

  test('it should show default text', () => {
    const tree = renderer.create(<RemoveButton />).toJSON();
    expect(tree).toHaveStyleRule('content', "'Remove All'", {
      modifier: ':after'
    });
  });

  test('it should accept text prop', () => {
    const tree = renderer.create(<RemoveButton text="Test Text" />).toJSON();
    expect(tree).toHaveStyleRule('content', 'Test Text', {
      modifier: ':after'
    });
  });
});
