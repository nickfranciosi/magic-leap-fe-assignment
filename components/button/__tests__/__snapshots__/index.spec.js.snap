// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton it should render 1`] = `
.c0 {
  background-color: transparent;
  border: 1px solid #edec51;
  color: #edec51;
  font-size: 16px;
  text-transform: uppercase;
  font-family: 'Lato',Helvetica,arial,sans-serif;
  padding: 16px;
  -webkit-letter-spacing: -1.1px;
  -moz-letter-spacing: -1.1px;
  -ms-letter-spacing: -1.1px;
  letter-spacing: -1.1px;
  -webkit-transition: background-color 200ms ease-in,color 200ms ease-in;
  transition: background-color 200ms ease-in,color 200ms ease-in;
  cursor: pointer;
}

.c0:hover {
  background-color: #edec51;
  color: #000;
}

<button
  className="c0"
/>
`;

exports[`RemoveButton it should render 1`] = `
.c0 {
  background: transparent;
  border: none;
  color: #e15021;
  position: absolute;
  top: 5px;
  left: 5px;
  z-index: 2;
  font-family: 'Lato',Helvetica,arial,sans-serif;
  font-size: 16px;
  text-transform: uppercase;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  cursor: pointer;
}

.c0:after {
  content: 'Remove All';
  display: inline;
  opacity: 0;
  margin-left: 4px;
  font-size: 0.75em;
  -webkit-transition: opacity 350ms ease;
  transition: opacity 350ms ease;
}

.c0:hover:after {
  opacity: 1;
}

<button
  className="c0"
/>
`;
